src/libraries/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -441,21 +446,85 @@ private class ClassWithInvalidDictionary
         }
 
         [Fact]
-        public static void ClassWithUnsupportedCollectionTypes()
+        public static void ClassWithUnsupportedArray()
         {
-            Exception ex;
+            Exception ex = Assert.Throws<NotSupportedException>(() =>
+                JsonSerializer.Deserialize<ClassWithInvalidArray>(@"{""UnsupportedArray"":[]}"));
```

Comment:
Yes per discussion the root type would always have "$" as the path and also the Type information would just be the `<T>` passed in (or `returnType` argument) so it is not normally useful.

Adding the extra `try\catch` is possible but does add a little cost in terms of performance and makes the code a bit more difficult to follow since now we have just a single `try\catch` and wrapping it would cause at least 3 more.